Replicator/Puller.cc

```diff
@@ -280,17 +285,17 @@ namespace litecore { namespace repl {
 
 
     void Puller::revReRequested(fleece::Retained<IncomingRev> inc) {
```

Comment:
This is an async method, and all it should do is enqueue a call to the corresponding private method. It should not have other logic like pulling the body size out of the IncomingRev. Instead, add a `bodySize` parameter.

Reply:
Not totally getting it. I have changed Puller:_revReRequested() to take parameter of, instead of inc, bodySize, which is all that _revReRequested() needs to do its job. The bodySize is store in inc object, and I pull it out as I enqueue it rather than as it is executed in the Actor's thread.

Reply:
Actor objects have an idiom of a public method that calls `enqueue` to schedule a call to a private method.
In this idiom the public method is simply a wrapper. The two methods have _the same parameter list_, except as necessary to manage ref-counting. The public method has no other logic in it.

In this case you have changed the parameter type of the public method but not the private method, and you have hidden some logic in the public method. This is very unlike the idiom and will be confusing.

You should change the _public_ `revReRequested` method to take the bodySize and simply pass it through to the private method. Then the caller, IncomingRev, does the work of finding the body size. This also means you don't have to expose a public method to access the body size.

Reply:
Got it. Will follow the idiom